server/service/orbit.go

```diff
@@ -310,18 +310,44 @@ func (svc *Service) GetOrbitConfig(ctx context.Context) (fleet.OrbitConfig, erro
 
 // filterExtensionsByPlatform filters a extensions configuration depending on the host platform.
 // (to not send extensions targeted to other operating systems).
-func filterExtensionsByPlatform(extensions json.RawMessage, hostPlatform string) (json.RawMessage, error) {
+//
+// If all extensions are filtered, then it returns (nil, nil) (Orbit expects empty extensions if there
+// are no extensions for the host.)
+func (svc *Service) filterExtensionsForHost(ctx context.Context, extensions json.RawMessage, host *fleet.Host) (json.RawMessage, error) {
 	if len(extensions) == 0 {
-		return extensions, nil
+		return nil, nil
 	}
 	var extensionsInfo fleet.Extensions
```

Comment:
Nit: This was a bit hard to follow for me because extensionsInfo is a map with ExtensionInfo values, which were iterating over using var extensionInfo